src/reducers/notifications.ts

```diff
@@ -0,0 +1,52 @@
+import { NotifyClientTypes } from '@walletconnect/notify-client'
+
+export interface TopicNotificationsState {
+  fullNotifications: NotifyClientTypes.NotifyMessage[]
+  existingIds: Set<string>
+  hasMore: boolean
+}
+
+export interface NotificationsState {
+  [topic: string]: TopicNotificationsState
+}
+
+export type NotificationsActions = {
+  type: 'FETCH_NOTIFICATIONS'
+  notifications: NotifyClientTypes.NotifyMessage[]
+  topic: string
+  hasMore: boolean
+}
+
+// Opted for a reducer since the state changes are complex enough to warrant
+// changes to a set and an array and having all that inside the hooks would
+// cause too many updates to the hooks, causing unnecessary rerenders.
+export const notificationsReducer = (
+  state: NotificationsState,
+  action: NotificationsActions
+): NotificationsState => {
+  const topicState = state[action.topic] as TopicNotificationsState | undefined
+
+  switch (action.type) {
+    case 'FETCH_NOTIFICATIONS':
+      const ids = topicState?.existingIds || new Set<string>()
+      const filteredVals = action.notifications.filter(val => !ids.has(val.id))
+      const uniqueIds = action.notifications.map(notification => notification.id)
```

Comment:
Aren't these all incoming ids? What makes them unique at this stage before being added to a `Set`?

Reply:
Ah, yes. Correct. Renaming